core/src/actors/session/handlers.rs

```diff
@@ -207,6 +214,24 @@ fn peer_discovery_peers(peers: &[Address]) {
     });
 }
 
+/// Function to process an Inv message
+fn inventory_process_inv(session: &mut Session, _ctx: &mut Context<Session>, inv: &Inv) {
+    // Check how many of the received inventory vectors need to be requested
+    let inv_vectors = &inv.inventory;
+
+    // TODO missing check of how many of these items really need to be requested
```

Comment:
Agree. Items already known to us can be just filtered out from the list. TODO.

Reply:
I'm working on it right now but I'm not sure if it will be on time for today's review :(. Would you be ok with approving it as it is? If I finish this functionality this morning, I'll create another PR.

Reply:
That's why I explicitly said TODO :stuck_out_tongue_winking_eye: